src/conversations/Conversation.ts

```diff
@@ -146,7 +146,7 @@ export class ConversationV2 {
     options?: SendOptions
   ): Promise<MessageV2> {
     const msg = await this.encodeMessage(message, options)
-    this.client.publishEnvelopes([
+    await this.client.publishEnvelopes([
```

Comment:
@mkobetic I also noticed that we weren't awaiting the sending of V2 messages.

This might explain why we needed to sleep for longer than expected to deal with race conditions in tests.

Reply:
Ah good catch. I wonder how many of these landmines have I buried in here. :/